src/Model/AsListIterator.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace MongoDB\Model;
+
+use IteratorIterator;
+use Traversable;
+
+/**
+ * @internal
+ * @template TKey as int
+ * @template TValue
+ * @template TIterator as Traversable<TKey, TValue>
+ *
+ * @template-extends IteratorIterator<TKey, TValue, TIterator>
+ */
+final class AsListIterator extends IteratorIterator
```

Comment:
This iterator is used when serialising `LazyBSONArray` instances to ensure that we return a packed array. The psalm template annotations may need some work (or silencing of errors), but for the time being this works.